src/record_syscall.cc

```diff
@@ -2330,7 +2329,12 @@ static Switchable prepare_ptrace(RecordTask* t,
       syscall_state.emulate_result(0);
       break;
     }
-    case PTRACE_OLDSETOPTIONS:
+    case X86Arch::PTRACE_OLDSETOPTIONS:
+      if (t->arch() != x86 && t->arch() != x86_64) {
+        syscall_state.emulate_result(-EINVAL);
```

Comment:
ptrace returns EIO for invalid requests.

Reply:
Ah, right. Good catch, thanks.